File: android/app/src/main/java/com/windchaser/runningos/domain/usecase/RunningUseCases.kt
package com.windchaser.runningos.domain.usecase

import com.windchaser.runningos.domain.model.UserStats
import com.windchaser.runningos.domain.repository.RunningRepository
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

/**
 * 获取用户统计数据用例
 */
class GetUserStatsUseCase @Inject constructor(
    private val repository: RunningRepository
) {
    operator fun invoke(): Flow<UserStats> {
        return repository.getUserStats()
    }
}

/**
 * 获取最新活动用例
 */
class GetLatestActivityUseCase @Inject constructor(
    private val repository: RunningRepository
) {
    operator fun invoke() = repository.getLatestActivity()
}

/**
 * 获取所有路线用例
 */
class GetAllRoutesUseCase @Inject constructor(
    private val repository: RunningRepository
) {
    operator fun invoke() = repository.getAllRoutes()
}

/**
 * 同步数据用例
 */
class SyncDataUseCase @Inject constructor(
    private val repository: RunningRepository
) {
    suspend operator fun invoke(): Result<Unit> {
        return repository.syncData()
    }
}

File: ios/RunningOS/Domain/UseCases/RunningUseCases.swift
import Foundation
import Combine

/// 获取用户统计数据用例
final class GetUserStatsUseCase {
    private let repository: RunningRepository

    init(repository: RunningRepository) {
        self.repository = repository
    }

    func execute() -> AnyPublisher<UserStats, Never> {
        return repository.getUserStats()
    }
}

/// 获取最新活动用例
final class GetLatestActivityUseCase {
    private let repository: RunningRepository

    init(repository: RunningRepository) {
        self.repository = repository
    }

    func execute() -> AnyPublisher<RunningActivity?, Never> {
        return repository.getLatestActivity()
    }
}

/// 获取所有路线用例
final class GetAllRoutesUseCase {
    private let repository: RunningRepository

    init(repository: RunningRepository) {
        self.repository = repository
    }

    func execute() -> AnyPublisher<[Route], Never> {
        return repository.getAllRoutes()
    }
}

/// 同步数据用例
final class SyncDataUseCase {
    private let repository: RunningRepository

    init(repository: RunningRepository) {
        self.repository = repository
    }

    func execute() -> AnyPublisher<Void, Error> {
        return repository.syncData()
    }
}
